### ISSUE TITLE
Adding a new user to MQTT Broker Add-on seems to disable the "homeassistant" user

### ISSUE THREAD

**Issue by donburch888** [home-assistant/addons#2223]
## The problem

Adding a new user to MQTT Broker Add-on seems to disable the "homeassistant" user

## Environment

- Add-on with the issue:  Mosquitto broker Add-on
- Add-on release with the issue: Current version: 6.0.1
- Last working add-on release (if known): 
- Operating environment (OS/Supervised): OS

## Problem-relevant configuration

```
  logins: 
    - username: user
      password: passwd
  customize:
    active: false
    folder: mosquitto
  certfile: fullchain.pem
  keyfile: privkey.pem
  require_certificate: false
```

## Traceback/Error logs

```
In the MQTT Broker Add-on's Log 
   1633389875: New connection from 172.30.32.1 on port 1883.
   1633389875: Socket error on client <unknown>, disconnecting.
   1633389995: New connection from 172.30.32.1 on port 1883.
   401: Unauthorized1633389996: Socket error on client <unknown>, disconnecting.
   1633390116: New connection from 172.30.32.1 on port 1883.
   1633390116: Socket error on client <unknown>, disconnecting.
   1633390236: New connection from 172.30.32.1 on port 1883.

Also errors is Log Viewer
   2021-10-05 10:38:37 WARNING (Thread-3) [homeassistant.components.mqtt] Disconnected from MQTT server core-mosquitto:1883 (5)
   2021-10-05 10:40:37 ERROR (Thread-3) [homeassistant.components.mqtt] Unable to connect to the MQTT broker: Connection Refused: not authorised.

and under "Supervisor > System" the Supervisor Log includes 
   21-10-05 10:32:36 INFO (MainThread) [supervisor.auth] Auth request from 'core_mosquitto' for 'homeassistant'
   21-10-05 10:32:37 WARNING (MainThread) [supervisor.auth] Unauthorized login for 'homeassistant'
   21-10-05 10:38:37 INFO (MainThread) [supervisor.auth] Auth request from 'core_mosquitto' for 'homeassistant'
   21-10-05 10:38:37 WARNING (MainThread) [supervisor.auth] Unauthorized login for 'homeassistant'
```

## Additional information

If we need to also add user "homeassistant" as well as our new user, what password to give for homeassistant ?

**Comment by donburch888** [comment]
over in the [Home Assistant Forum](https://community.home-assistant.io/t/home-assistant-mosquitto-tasmota-device-socket-error/344970), moderator Tom I said:

> As well as the Mosquitto addon, you have to add the Home Assistant MQTT integration (so Home Assistant can talk to the MQTT addon). Have you done this?
> 
> If not you can find in the Configuration / Integrations page, then the Add Integration button (lower right).
> 
> Also do keep the addon user/password config and make sure the user is not ‘homeassistant’. Anonymous logins are no longer supported.

Sure enough, the MQTT HA Integration is where user homeassistant is defined and password given. Seems like this is something to add to the MQTT Add-on's documentation. 

Copying these into the MQTT Add-On's config did solve the problem - at least for a while...

The MQTT Add-on's log shows good connections from Rhasspy base station and bothe satellites ... then back to the disconnections:
```
1633778400: New connection from 192.168.1.93 on port 1883.
1633778400: New connection from 192.168.1.93 on port 1883.
1633778400: New connection from 192.168.1.93 on port 1883.
1633778400: New client connected from 192.168.1.93 as auto-6365F235-E57F-A825-6D64-367217E9BB49 (p2, c1, k60, u'rhasspy').
1633778400: New connection from 192.168.1.93 on port 1883.
1633778400: New connection from 192.168.1.93 on port 1883.
1633778400: New client connected from 192.168.1.93 as auto-F4852A9A-E42C-857C-72B3-7DD67648EEF2 (p2, c1, k60, u'rhasspy').
1633778400: New client connected from 192.168.1.93 as auto-4F522BEA-82A5-FFFC-AA8B-7E6E50FCB76C (p2, c1, k60, u'rhasspy').
1633778400: New client connected from 192.168.1.93 as auto-A5F10030-2623-C2AB-B5C1-A00F9A8E523A (p2, c1, k60, u'rhasspy').
1633778400: New client connected from 192.168.1.93 as auto-F5C6122D-9BCD-858D-C637-BFF7D1139170 (p2, c1, k60, u'rhasspy').
1633778489: New connection from 192.168.1.94 on port 1883.
401: Unauthorized1633778489: Socket error on client <unknown>, disconnecting.
1633778508: New connection from 172.30.32.1 on port 1883.
401: Unauthorized1633778509: Socket error on client <unknown>, disconnecting.
1633778609: New connection from 192.168.1.94 on port 1883.
1633778609: Socket error on client <unknown>, disconnecting.
1633778629: New connection from 172.30.32.1 on port 1883.
1633778629: Socket error on client <unknown>, disconnecting.
1633778730: New connection from 192.168.1.94 on port 1883.
1633778730: Socket error on client <unknown>, disconnecting.
```

Restarting the MQTT Add-on adds the following to the Supervisor > System Supervisor log
```
21-10-09 22:19:52 INFO (SyncWorker_1) [supervisor.docker.interface] Stopping addon_core_mosquitto application
21-10-09 22:19:54 INFO (SyncWorker_1) [supervisor.docker.interface] Cleaning addon_core_mosquitto application
21-10-09 22:19:56 INFO (SyncWorker_3) [supervisor.docker.addon] Starting Docker add-on homeassistant/aarch64-addon-mosquitto with version 6.0.1
21-10-09 22:20:00 ERROR (MainThread) [supervisor.services.modules.mqtt] There is already a MQTT service in use from core_mosquitto
21-10-09 22:20:00 INFO (MainThread) [supervisor.auth] Auth request from 'core_mosquitto' for 'don'
21-10-09 22:20:01 INFO (MainThread) [supervisor.auth] Successful login for 'don'
21-10-09 22:21:29 INFO (MainThread) [supervisor.auth] Auth request from 'core_mosquitto' for 'rhasspy'
21-10-09 22:21:29 WARNING (MainThread) [supervisor.auth] Unauthorized login for 'rhasspy'
21-10-09 22:21:48 INFO (MainThread) [supervisor.auth] Auth request from 'core_mosquitto' for 'homeassistant'
21-10-09 22:21:49 WARNING (MainThread) [supervisor.auth] Unauthorized login for 'homeassistant'
21-10-09 22:22:13 WARNING (MainThread) [supervisor.host.network] Can't update connectivity information: Error: Timeout was reached
21-10-09 22:27:30 INFO (MainThread) [supervisor.auth] Auth request from 'core_mosquitto' for 'rhasspy'
21-10-09 22:27:30 WARNING (MainThread) [supervisor.auth] Unauthorized login for 'rhasspy'
21-10-09 22:27:49 INFO (MainThread) [supervisor.auth] Auth request from 'core_mosquitto' for 'homeassistant'
21-10-09 22:27:49 WARNING (MainThread) [supervisor.auth] Unauthorized login for 'homeassistant'
```
then in the log viewer
```
2021-10-09 22:25:49 ERROR (Thread-3) [homeassistant.components.mqtt] Unable to connect to the MQTT broker: Connection Refused: not authorised.
2021-10-09 22:25:49 WARNING (Thread-3) [homeassistant.components.mqtt] Disconnected from MQTT server core-mosquitto:1883 (5)
2021-10-09 22:27:49 ERROR (Thread-3) [homeassistant.components.mqtt] Unable to connect to the MQTT broker: Connection Refused: not authorised.
2021-10-09 22:27:49 WARNING (Thread-3) [homeassistant.components.mqtt] Disconnected from MQTT server core-mosquitto:1883 (5)
2021-10-09 22:27:57 ERROR (MainThread) [frontend.js.latest.202108300] :0:0 ResizeObserver loop completed with undelivered notifications.
```

is it possible there is more than one mosquitto broker running on the HA/Rhaspy base station ? 

What else can I look at / do to help debug ? The entries being logged aren't providing enough info to be very helpful ;-(

**Comment by donburch888** [comment]
> If we need to also add user "homeassistant" as well as our new user, what password to give for homeassistant ?

Found it ! Under Configure > Integrations > Mosquitto Broker > Configure > Re-Configure MQTT you will find the user "homeassistant" and its password. 

I copied homeassistant's password, then went to Configuration > Users and [+ Add user] to add "homeassistant" and the new user I had initially tried to add for the Rhasspy sateliites. Return the HA Add-on MQTT Broker's configuration to "logins: []"

One day later, and i am no longer seeing the 401 errors or most of the Socket errors.
